### PR TITLE
Bold messages to make it clearer, keep messages when undoing

### LINKED ISSUE #185: Suggestion: in-game chat either bold/different colour then in-game log
Finding the in-game chat quite handy and useful. If possible it would be nice to have it different font/typeface than the in-game log text.

Just a nice to have/suggestion - definitely not a complaint! This site is coming along so nicely, great work!
Rob Thompson

### PR DESCRIPTION
**Pull request by jenf**
Implements two of the suggestions
* Bold messages
* Retain messages when a user presses the undo button.

resolves #185

### CODE REVIEW

File: assets/app/view/log.rb [hunk 1]
```diff
@@ -26,7 +26,11 @@ def render
       }
 
       lines = @log.reverse.map do |line|
-        h(:div, { style: { transform: 'scaleY(-1)' } }, line)
+        if line.is_a? String
```
Comment (tobymao) [inline assets/app/view/log.rb:29]: always use parens

File: lib/engine/game/base.rb [hunk 2]
```diff
@@ -225,7 +225,8 @@ def clone(actions)
       end
 
       def rollback
-        clone(@actions[0...-1])
+        action_to_remove = @actions.rindex { |x| !x.keep_on_rollback? }
+        clone(@actions[0, action_to_remove] + @actions[action_to_remove + 1..-1])
```
Comment (tobymao) [inline lib/engine/game/base.rb:229]: @actions.slice...

**Comment by tobymao** [comment]
have you tested this on a running instance? i think this will break the sorting in the database

actions have IDS and those are persisted. i think doing this is more complicated since you need to preserve relative order of the messages. not sure we want to do this.

**Comment by jenf** [comment]
Your right, i'd tried undoing and it says it's the wrong turn.

But I think making messages persist is important especially in async games where you can't contact people another way.

**Comment by jenf** [comment]
Closing as undo will be more complicated, separate PR for bold raised

**Pull request closed without merging**